docs/rules/0191/csharp-namespace.md

```diff
@@ -0,0 +1,72 @@
+---
+rule:
+  aip: 191
+  name: [core, '0191', csharp-namespace]
+  summary: The `option csharp_namespace` annotation should be idiomatic if set.
+permalink: /191/java-package
+redirect_from:
+  - /0191/csharp-namespace
+---
+
+# C# namespace annotation
+
+This rule enforces that if a proto file for a public API surface sets
+`option csharp_namespace`, that it uses language idiomatic conventions, as
+mandated in [AIP-191][].
+
+## Details
+
+This rule looks at each proto file, and complains if the `csharp_namespace`
+file annotation uses anything other than upper camel case, or includes
+characters other than letters, numbers, and `.`.
+
+## Examples
```

Comment:
Could we have another example with a beta? This is a common error - using V1beta instead of V1Beta.

Reply:
Done.